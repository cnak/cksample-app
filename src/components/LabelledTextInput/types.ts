import { TextInputProps } from 'react-native';

import { ValidationError } from '../ErrorIsland/types';

export interface LabelledTextInputProps extends TextInputProps {
    label?: string;
    hint?: string;
    width?: string;
    id?: string;
    children?: any;
    showBorderOnError?: boolean;
    showLabel?: boolean;
    validationField?: string;
    errors?: Array<ValidationError>;
}
