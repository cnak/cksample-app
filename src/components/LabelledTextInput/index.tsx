import React from 'react';

import TextInput from '../TextInput';
import { ErrorLabel, Label } from '../Typography';
import ValidationView from '../ValidationView';
import { BLACK, RED } from '../colors';

import { LabelledTextInputProps } from './types';

const DEFAULT_WIDTH = '100%';

export const LabelledTextInput = (props: LabelledTextInputProps) => {
    const { label, errors, showBorderOnError, showLabel, validationField, width } = props;

    const error = errors && errors.length > 0 ? errors.find((e) => e.field === validationField) : undefined;
    const borderColor = error ? RED : BLACK;
    const errorMessage = error && error.message;
    return (
        <ValidationView hasError={!!error} showBorderOnError={showBorderOnError}>
            {label && showLabel && <Label accessibilityLabel={`Label for input. ${label}`}>{label}</Label>}
            {!!errorMessage && <ErrorLabel accessibilityLabel={`Error. ${errorMessage}`}>{errorMessage}</ErrorLabel>}
            <TextInput
                accessibilityLabel={`Input. ${label}`}
                width={width || DEFAULT_WIDTH}
                style={{ borderColor }}
                autoCompleteType="off"
                {...props}
            />
        </ValidationView>
    );
};

LabelledTextInput.defaultProps = {
    showLabel: true
};

export default LabelledTextInput;
