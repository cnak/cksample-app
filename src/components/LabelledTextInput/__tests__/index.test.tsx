import React from 'react';
import renderer from 'react-test-renderer';

import LabelledTextInput from '../.';
describe('<LabelledTextInput>', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<LabelledTextInput label="Username" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
