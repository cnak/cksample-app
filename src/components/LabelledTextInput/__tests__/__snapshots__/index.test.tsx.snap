// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelledTextInput> matches snapshot 1`] = `
<View
  hasError={false}
  showBorderOnError={true}
  style={
    Array [
      Object {
        "borderLeftColor": "#0B0C0C",
        "borderLeftWidth": 0,
        "marginBottom": 25,
        "paddingLeft": 0,
      },
    ]
  }
>
  <Text
    accessibilityLabel="Label for input. Username"
    style={
      Array [
        Object {
          "color": "#0B0C0C",
          "fontFamily": "Helvetica",
          "fontSize": 16,
          "lineHeight": 20,
          "marginBottom": 10,
        },
      ]
    }
  >
    Username
  </Text>
  <View
    showFocus={false}
    style={
      Array [
        Object {
          "borderColor": "transparent",
          "borderWidth": 3,
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <TextInput
      accessibilityLabel="Input. Username"
      allowFontScaling={true}
      autoCompleteType="off"
      autoCorrect={false}
      label="Username"
      onBlur={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      showLabel={true}
      spellCheck={false}
      style={
        Array [
          Object {
            "borderColor": "black",
            "borderStyle": "solid",
            "borderWidth": 2,
            "fontSize": 16,
            "height": 40,
            "lineHeight": 20,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "width": "100%",
          },
          Object {
            "borderColor": "#0B0C0C",
          },
        ]
      }
      testID="TextInput"
      underlineColorAndroid="transparent"
      width="100%"
    />
  </View>
</View>
`;
